feat(VHotkey): add VHotkey component

MatthewAry opened the pull request:
- Introduced VHotkey component to render keyboard shortcuts
- Added localization entries for hotkey terms in multiple languages.
- Created JSON files for hotkey localization and updated auto-imports for new components.
- Included styles for the VHotkey component
- Enhanced documentation for the new component in the auto-imports and relevant markdown files.

<!--
MAKE SURE TO READ THE CONTRIBUTING GUIDE BEFORE CREATING A PR
https://vuetifyjs.com/getting-started/contributing

Provide a general summary of your changes in the title above
Keep the title short and descriptive, as it will be used as a commit message
PR titles should follow conventional-changelog-angular:
https://vuetifyjs.com/getting-started/contributing/#commit-guidelines
-->

## Description
<!--
Describe your changes in detail. Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
e.g. resolves #4213 or fixes #2312
-->

## Markup:
<!--
Information on how to set up your local development environment can be found here:
https://vuetifyjs.com/getting-started/contributing/#setting-up-your-environment
Remove this section for documentation or test-only changes.
-->

<!-- Paste your FULL packages/vuetify/dev/Playground.vue here --->
```vue

```

Haviles04 commented on packages/vuetify/src/labs/VHotkey/VHotkey.tsx:
```diff
@@ -0,0 +1,401 @@
+    const keyConfig = (isMac && config.mac) ? config.mac : config.default
+    const value = keyConfig[mode] ?? keyConfig.text
+
+    // If we requested icon mode but no icon is available, fallback to text mode
+    if (mode === 'icon' && !keyConfig.icon) {
+      return ['text', value]
+    }
+
+    // If we requested symbol mode but no symbol is available, fallback to text mode
+    if (mode === 'symbol' && !keyConfig.symbol) {
+      return ['text', value]
+    }
```
Can these be combined into one? 


```suggestion
 
    if (!keyConfig.[mode]) {
      return ['text', value]
    }
```

MatthewAry replied:
The current code structure is:

```typescript
type KeyConfig = {
  symbol?: string
  icon?: string
  text: string
}
```

Notice that:
1. `symbol` and `icon` are **optional** properties (marked with `?`)
2. `text` is **required** 
3. The `value` variable is set to `keyConfig[mode] ?? keyConfig.text`

Your suggested change would have these problems:

1. If `mode` is `'text'`, then `keyConfig[mode]` would be the text value (which is always present), so the condition would never be true for text mode.
2. The current code specifically handles the cases where:
   - They request `'icon'` mode but no icon is defined → fallback to text
   - They request `'symbol'` mode but no symbol is defined → fallback to text
   - For `'text'` mode, it always uses the text value (no fallback needed)
3. The current approach ensures that when an icon or symbol isn't available, it falls back to text mode.

The existing code is more explicit and handles the three different modes correctly. It could potentially be simplified to:

```typescript
if ((mode === 'icon' && !keyConfig.icon) || (mode === 'symbol' && !keyConfig.symbol)) {
  return ['text', value]
}
```

But I think the current version is easier to understand.

Haviles04 reviewed (commented):
Cool stuff, 1 small nitpick :)

MatthewAry commented:
`packages/docs/src/examples/v-hotkey/prop-key-map.vue` has some pretty dumb lint errors. Not sure what to do about that. I fixed bugs with platform handling and added a new prop to VHotkey for DX, Testing, and Documentation purposes.

J-Sek commented on packages/vuetify/src/components/VKbd/VKbd.sass:
```diff
@@ -3,11 +3,15 @@
+    justify-content: center
+    min-width: $kbd-min-width
     padding: $kbd-padding
 
```
`line-height: 1` could help avoid inconsistent height

![image](https://github.com/user-attachments/assets/5e8e8320-f2a7-4de1-bdf1-5724e9454715)

MatthewAry replied:
I agree about need for improved height consistency. Specifically with symbol mode vs others. I would love to hear your thoughts after you review the latest changes.

J-Sek commented on packages/vuetify/src/labs/VHotkey/VHotkey.tsx:
```diff
@@ -0,0 +1,430 @@
+  }),
+  escape: createKey({
+    default: { text: '$vuetify.hotkey.escape' },
+  }),
```
I think we are missing `space` here. There is a dedicated icon `mdi-keyboard-space` ready to use.

MatthewAry replied:
Yes, we could add space. But do we really need/want to? I personally hate the fact that Macs have symbols on the keyboard to begin with.

MatthewAry replied:
I have already added a title property to icon/symbol representations such that they should be able to see the translation of the icon. I can add the space icon. I just think it's stupid. I wonder if I should change the way the keymap is exposed through the prop such that people would be able to exclude icons/symbols. Right now, they can only modify or add.

J-Sek commented on packages/vuetify/src/labs/VHotkey/VHotkey.scss:
```diff
@@ -0,0 +1,106 @@
+      align-items: baseline;
+      max-height: 1lh;
+      vertical-align: baseline;
+      font-size: $hotkey-inline-font-size;
```
dividers within `--inline` ends up with 10.5px font size. Should they be set to 1em as well? Perhaps with a dedicated Sass variable?

MatthewAry commented:
I updated my playground!

MatthewAry commented:
![image](https://github.com/user-attachments/assets/f802a27f-374e-4686-9bbe-d78ef91e6283)
Looks pretty good!

J-Sek commented on packages/vuetify/src/iconsets/mdi-svg.ts:
```diff
@@ -48,6 +48,16 @@ const aliases: IconAliases = {
   upload: 'svg:M11 20H6.5q-2.28 0-3.89-1.57Q1 16.85 1 14.58q0-1.95 1.17-3.48q1.18-1.53 3.08-1.95q.63-2.3 2.5-3.72Q9.63 4 12 4q2.93 0 4.96 2.04Q19 8.07 19 11q1.73.2 2.86 1.5q1.14 1.28 1.14 3q0 1.88-1.31 3.19T18.5 20H13v-7.15l1.6 1.55L16 13l-4-4l-4 4l1.4 1.4l1.6-1.55Z',
   color: 'svg:M17.5 12a1.5 1.5 0 0 1-1.5-1.5A1.5 1.5 0 0 1 17.5 9a1.5 1.5 0 0 1 1.5 1.5a1.5 1.5 0 0 1-1.5 1.5m-3-4A1.5 1.5 0 0 1 13 6.5A1.5 1.5 0 0 1 14.5 5A1.5 1.5 0 0 1 16 6.5A1.5 1.5 0 0 1 14.5 8m-5 0A1.5 1.5 0 0 1 8 6.5A1.5 1.5 0 0 1 9.5 5A1.5 1.5 0 0 1 11 6.5A1.5 1.5 0 0 1 9.5 8m-3 4A1.5 1.5 0 0 1 5 10.5A1.5 1.5 0 0 1 6.5 9A1.5 1.5 0 0 1 8 10.5A1.5 1.5 0 0 1 6.5 12M12 3a9 9 0 0 0-9 9a9 9 0 0 ...
+  command: 'svg:M6 2a4 4 0 0 1 4 4v2h4V6a4 4 0 0 1 4-4a4 4 0 0 1 4 4a4 4 0 0 1-4 4h-2v4h2a4 4 0 0 1 4 4a4 4 0 0 1-4 4a4 4 0 0 1-4-4v-2h-4v2a4 4 0 0 1-4 4a4 4 0 0 1-4-4a4 4 0 0 1 4-4h2v-4H6a4 4 0 0 1-4-4a4 4 0 0 1 4-4m10 16a2 2 0 0 0 2 2a2 2 0 0 0 2-2a2 2 0 0 0-2-2h-2zm-2-8h-4v4h4zm-8 6a2 2 0 0 0-2 2a2 2 0 0 0 2 2a2 2 0 0 0 2-2v-2zM8 6a2 2 0 0 0-2-2a2 2 0 0 0-2 2a2 2 0 0 0 2 2h2zm10 2a2 2 0 0 0 2- ...
+  ctrl: 'svg:m19.78 11.78l-1.42 1.41L12 6.83l-6.36 6.36l-1.42-1.41L12 4z',
```
Not sure if it matters, but I would copy those straight from `@mdi/js`

MatthewAry replied:
They came from https://icones.js.org/ record of those icons. Will update from OG reference

J-Sek commented on packages/vuetify/src/labs/VHotkey/VHotkey.tsx:
```diff
@@ -0,0 +1,551 @@
+    type: Object as PropType<KeyMap>,
+    default: keyMap,
+  },
+  overridePlatform: {
```
How about instead of `overridePlatform` make it into `platform: 'auto' | 'pc' | 'mac'` with default `'auto'`?

MatthewAry replied:
Great Idea. Will introduce that after this up-coming push.

J-Sek commented on packages/vuetify/src/labs/VHotkey/VHotkey.scss:
```diff
@@ -0,0 +1,232 @@
+    }
+
+    &--variant-flat &__key.v-kbd {
+      @include tools.elevation(0);
```
Those are not that unique in our codebase (found 3 other components calling that helper with `0`), but I am not sure if it brings any value over `box-shadow: none`.

MatthewAry replied:
Good idea.

MatthewAry replied:
Will do

MatthewAry commented on packages/vuetify/src/util/hotkey-parsing.ts:
```diff
@@ -0,0 +1,184 @@
+  flushBuffer()
+
+  // Combinations like `ctrl-b` are not valid, `-` is a sequence separator.
+  // `-` is only a valid key if it's on its own.
```
This comment is wrong! This should be a valid combo! I blame AI.
